src/ringct/rctTypes.cpp

```diff
@@ -118,39 +119,21 @@ namespace rct {
     //uint long long to 32 byte key
     void d2h(key & amounth, const xmr_amount in) {
         sc_0(amounth.bytes);
-        xmr_amount val = in;
-        int i = 0;
-        while (val != 0) {
-            amounth[i] = (unsigned char)(val & 0xFF);
-            i++;
-            val /= (xmr_amount)256;
-        }
+        memcpy_swap64le(amounth.bytes, &in, 1);
     }
     
     //uint long long to 32 byte key
     key d2h(const xmr_amount in) {
         key amounth;
-        sc_0(amounth.bytes);
-        xmr_amount val = in;
-        int i = 0;
-        while (val != 0) {
-            amounth[i] = (unsigned char)(val & 0xFF);
-            i++;
-            val /= (xmr_amount)256;
-        }
+        d2h(amounth, in);
         return amounth;
     }
 
     //uint long long to int[64]
     void d2b(bits  amountb, xmr_amount val) {
         int i = 0;
-        while (val != 0) {
-            amountb[i] = val & 1;
-            i++;
-            val >>= 1;
-        }
         while (i < 64) {
-            amountb[i] = 0;
+            amountb[i] = (val >> i) & 1;
```

Comment:
This is only constant time if a particular CPU has a barrel shifter. Otherwise it can be linear in `i`. For constant time it would be better to do 1-bit shifts every time, like so:
````
while (i < 64) {
  amountb[i] = val & 1;
  val >>=1;
}
````